src/components/panel/CrossRefLink.tsx

```diff
@@ -0,0 +1,168 @@
+
+    if (newContentIndex === currentContentIndex) {
+      const targetEl = panelEl.querySelector(targetSelector) as HTMLElement
+      targetEl.scrollIntoView({ behavior: 'smooth' })
+      return
+    }
+
+    // target is in new content type text
+    setTimeout(() => {
+      updatePanel({ contentIndex: newContentIndex })
+    }, 200)
+
+    const targetEl = panelEl.querySelector(targetSelector) as HTMLElement
+    targetEl.scrollIntoView({ behavior: 'smooth' })
+  }
+
+  async function scrollToDifferentItem(sourceEl, action, panelId) {
+    let newPanelId = panelId
+    const collectionId = sourceEl.getAttribute('data-ref-collection')
+    const manifestId = sourceEl.getAttribute('data-ref-manifest')
+    const itemId = sourceEl.getAttribute('data-ref-item')
+    const contentType = sourceEl.getAttribute('data-ref-content-type')
+    const selector = sourceEl.getAttribute('data-ref-target')
+
+    const collection = await useDataStore.getState().initCollection(collectionId)
+    const manifestIndex = collection.sequence.findIndex(m => m.id === manifestId)
+    const manifest = await apiRequest<Manifest>(collection.sequence[manifestIndex].id)
+    const itemIndex = manifest.sequence.findIndex(i => i.id === itemId)
+    const item = await apiRequest<Item>(itemId)
+    const newContentTypes = getContentTypes(item.content)
+    const newContentIndex = newContentTypes.findIndex(type => type === contentType)
+
+    if (action === 'new') {
+      const newPanelConfig = {
+        collection: collectionId,
+        manifestIndex,
+        itemIndex,
+        contentIndex: newContentIndex
+      }
+
+      newPanelId = crypto.randomUUID()
+      useUIStore.getState().updateNewestPanelId(newPanelId)
+      await usePanelStore.getState().addPanel(newPanelConfig, newPanelId)
+    }
+
+    updatePanel({
+      collectionId,
+      manifest: manifest,
+      item,
+      contentIndex: newContentIndex
+    })
+
+    // use setTimeout to create a small delay after update or creating a new panel
+    setTimeout(async () => {
+      waitForElementInDom('#' + newPanelId, (panelEl: HTMLElement) => {
+        const targetEl = panelEl.querySelector(selector) as HTMLElement
+        targetEl.scrollIntoView({ behavior: 'smooth' })
+      })
+    }, 500)
+  }
+
+  const link = <a ref={ref}
+    className={
+      (props.className || '') +
+        (isHighlighted ? ' bg-gray-200 relative cursor-pointer' : '') +
+        ' text-blue-600 underline cursor-pointer' +
+        (isHovered ? ' bg-primary/20' : '') +
+        (selectedAnnotation && selectedAnnotation.id === props['data-annotation'] ? 'bg-primary/40' : '')
+    }
+    onMouseEnter={handleMouseEnter}
+    onMouseLeave={handleMouseLeave}
+    onClick={!isTargetInDiffItem ? onClick : null}
+  >
+    {node.innerHTML}
+    {isTargetInDiffItem ? <ExternalLink size={16} className="inline ml-[1px]" /> : ''}
+  </a>
+
+
+  if (isTargetInDiffItem) return <DropdownMenu>
```

Comment:
so far so good, but to improve more the component design it would be better now to create two more child components CrossRefInternalLink and CrossRefExternalLink and use them like so `<CrossRefExternalLink>{node.innerHTML}<CrossRefExternalLink>`. Pass all the needed things as props to each of those and define the respective functions inside the right comp.